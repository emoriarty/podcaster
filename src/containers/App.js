import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import config from '../../config/app'
import AppBar from 'material-ui/AppBar'
import FlatButton from 'material-ui/FlatButton'
import Avatar from 'material-ui/Avatar'
import {
  fetchCommonTranslations,
  fetchCountries,
  fetchMediaTypes,
  fetchMediaTypesTranslations,
  fetchPodcasts
} from '../actions'
import { pleaseWait } from 'please-wait'
import logo from '../../assets/images/logo-256.png'
import 'please-wait/src/please-wait.scss'

export class App extends Component {
  componentDidMount () {
    const {
      fetchCommonTranslations,
      fetchCountries,
      fetchMediaTypes,
      fetchMediaTypesTranslations,
      language
    } = this.props

    this.loadingScreen = pleaseWait({
      logo,
      backgroundColor: '#1976D2',
      loadingHtml: '<div class="spinner"><div class="double-bounce1"></div><div class="double-bounce2"></div></div>'
    })

    fetchCommonTranslations(language)
    fetchCountries()
    fetchMediaTypes()
    fetchMediaTypesTranslations(language)
  }

  componentWillReceiveProps (nextProps) {
    const { country, fetchPodcasts, podcastUrl } = this.props
    if (!nextProps.isLoading) {
      this.loadingScreen.finish()
    }
    console.log(nextProps.podcastUrl)
    if (nextProps.podcastUrl !== podcastUrl) {
      fetchPodcasts(nextProps.podcastUrl, country, 'limit=10/genre=1303/explicit=true')
    }
  }

  renderFlagButton () {
    const { countryFlag } = this.props

    return (
      <FlatButton>
        <Avatar
          style={{
            backgroundColor: 'transparent',
            marginTop: 3
          }}
          size={30}
          src={countryFlag}
        />
      </FlatButton>
    )
  }

  render () {
    const { isLoading } = this.props
    return (
      <div>
        {
          !isLoading &&
            <div>
              <AppBar
                showMenuIconButton={false}
                title='Podcaster'
                iconElementRight={this.renderFlagButton()}
              />
              <h2>Top</h2>
              <ul>
                <li>podcast 1</li>
              </ul>
            </div>
        }
      </div>
    )
  }
}

App.defaultProps = {
  isLoading: true
}

App.propTypes = {
  country: PropTypes.string,
  countryFlag: PropTypes.string,
  fetchCommonTranslations: PropTypes.func,
  fetchCountries: PropTypes.func,
  fetchMediaTypes: PropTypes.func,
  fetchMediaTypesTranslations: PropTypes.func,
  fetchPodcasts: PropTypes.func,
  language: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  podcastUrl: PropTypes.string
}

const mapStateToProps = (state) => {
  const countryCode = state.settings.country
  const countryFlag = countryCode in state.entities.countries
    ? state.entities.countries[countryCode].flag_icon
    : ''

  return {
    country: state.settings.country,
    countryFlag: `${config.itunes.provider.rss}${countryFlag}`,
    podcastUrl: '26' in state.entities.mediaTypes ? state.entities.mediaTypes['26'].feed_types.urlPrefix : '',
    isLoading: state.countries.isFetching ||
      state.mediaTypes.isFetching ||
      state.translations.common.isFetching ||
      state.translations.mediaTypes.isFetching,
    language: state.settings.language
  }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    fetchCommonTranslations,
    fetchCountries,
    fetchMediaTypes,
    fetchMediaTypesTranslations,
    fetchPodcasts
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App)
