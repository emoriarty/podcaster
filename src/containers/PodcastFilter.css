.podcast-filter {
  margin: 0 auto 32px auto;
  width: calc(100% - 32px);
}

.select-field {
  width: 100% !important;
}

.toggle {
  margin-top: 16px;
}
