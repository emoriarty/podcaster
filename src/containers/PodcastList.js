import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { fetchPodcasts } from '../actions'
import { getPodcasts } from '../selectors/podcasts'
import { getMediaTypeUrl } from '../selectors/mediaTypes'
import { getCountry } from '../selectors/settings'
import times from '../utils/times'
import { List } from 'material-ui/List'
import {
  TwoLineListItem,
  TwoLineLoadingListItem
} from '../components/ListItems'

class PodcastList extends Component {
  static propTypes = {
    country: PropTypes.string,
    podcasts: PropTypes.array,
    podcastsUrl: PropTypes.string
  }

  constructor (props) {
    super(props)

    this.renderItems = this.renderItems.bind(this)
    this.renderLoadingItems = this.renderLoadingItems.bind(this)
  }

  componentWillReceiveProps (nextProps) {
    // const { country, fetchPodcasts, podcastsUrl } = this.props
    //
    // if (nextProps.podcastsUrl !== podcastsUrl || nextProps.country !== country) {
    //   fetchPodcasts(nextProps.podcastsUrl, country, 'limit=50/explicit=true')
    // }
  }

  renderLoadingItems () {
    let items = []
    times(10)(
      index =>
        items.push(<TwoLineLoadingListItem key={index} />)
    )
    console.log(items)
    return items
  }

  renderItems () {
    const { podcasts } = this.props
    return podcasts.map(
      podcast =>
        <TwoLineListItem
          key={podcast['id.attributes.im:id']}
          primaryText={podcast['im:name.label']}
          secondaryText={podcast['im:artist.label']}
          avatar={podcast['im:image.0.label']}
        />
    )
  }

  render () {
    const { podcasts } = this.props

    return (
      <List>
        {
          podcasts.length > 0
            ? this.renderItems()
            : this.renderLoadingItems()
        }
      </List>
    )
  }
}

const mapStateToProps = state => {
  const country = getCountry(state)
  const podcasts = getPodcasts(state)
  const podcastsUrl = getMediaTypeUrl(state, 'podcasts')

  return {
    country,
    podcasts,
    podcastsUrl
  }
}

const mapDispatchToProps = {
  fetchPodcasts
}

export default connect(mapStateToProps, mapDispatchToProps)(PodcastList)
