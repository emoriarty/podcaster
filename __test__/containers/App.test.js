import React from 'react'
import { App } from '../../src/containers/App'

describe('App container', () => {
  it('shows default home page', () => {
    const component = shallow(<App translations={{}}language='es-ES' />)
    expect(component)
      .toMatchSnapshot()
  })
})
