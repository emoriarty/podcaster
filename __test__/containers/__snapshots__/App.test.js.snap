// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container shows default home page 1`] = `
<div>
  <AppBar
    showMenuIconButton={false}
    title="Podcaster"
    zDepth={1}
  />
  <AppContainer>
    <Headline />
    <Connect(PodcastList) />
  </AppContainer>
</div>
`;
